DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,179 @@
+
+```
+./lint.sh
+```
+
+- By default, the script lints both Python and Node.js (if changes are staged), but this can be overridden with the `-p` and `-n` flags respectively.
+
+    ```
+    ./lint.sh -p false
+    ```
+
+- You may wish to symlink the script as a pre-commit hook to lint your changes before committing.
+
+    ```
+    ln -s ../../lint.sh .git/hooks/pre-commit
+    ```
+
+## Usage with preexisting databases
+
+If you want to use Mathesar with a preexisting Postgres DB, modify the `DATABASES.mathesar_tables` entry of the `config/settings.py` file with appropriate connection details before installing the Mathesar types and functions by running `install.py` as described in the previous step.
+
+## Rebuilding the Docker images
+
+Sometimes you may need to rebuild your Docker images after pulling new code changes or switching branches. Do so via:
+
+```
+docker compose --profile dev up --force-recreate --build dev-service
+```
+
+
+## Demo mode
+
+Mathesar can be run in "demo mode" to meet the specific needs of our [live demo site](https://demo.mathesar.org).
+
+See our [Live demo mode](./demo/README.md) guide for more information on enabling live demo mode locally
+
+
+## Opening a shell in the container
+
+- If you need to do some work within the container you can open a bash shell via:
+
+    ```
+    docker exec -it mathesar_service_dev bash
+    ```
+
+- To open a PostgreSQL [psql](https://www.postgresql.org/docs/current/app-psql.html) terminal for the data in Mathesar:
+
+    ```
+    docker exec -it mathesar_db psql -U mathesar
+    ```
+
+## Troubleshooting
+
+### Permissions within Windows
+
+- Running Script in powershell is disabled by default in windows , you have to change permission to run scripts  [Official Docs ](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-7.2) 
+
+### Fixing line endings on Windows
+
+If you happen to clone the git repository outside of WSL, then you fix it by running these commands from within WSL.
+
+```
+git config --global core.autocrlf input
+sudo apt-get install -y dos2unix
+sudo find . -type f -exec dos2unix {} \;
+```
+
+These commands install the `dos2unix` utility, which converts text files from the DOS/Microsoft Windows format (with CRLF line endings) to the Unix/Linux format (with LF line endings). Next, the find utility is used to locate all files (-type f) in the current directory (.) and its subdirectories, and the dos2unix command is then executed on each of them (-exec dos2unix ;).
+
+
+### Live reloading front end code on Windows
+
+Hot module replacement for front end code does not work when the project is present on a windows filesystem and WSL is used to run docker. [Read more](./mathesar_ui/README.md#live-reloading-on-windows).
+
+### PostgreSQL server is running on your host machine
+
+* If you you see the following error after running `docker-compose up`, then the port used by Postgres is already in use.
```

Comment:
`docker-compose up` may confuse people. I know what you mean here, but it's not actually the command used above.

Reply:
The content in this _"PostgreSQL server is running on your host machine"_ section is [unchanged](https://github.com/centerofci/mathesar-wiki/blob/41309a88b87a45f00f7d57bd5cffe59113876513/engineering/setup/common-issues.md?plain=1#L40L54) in this PR, but I agree it could use some cleanup. In this case the cleanup is more straightforward (than in the commit message case above) so I pushed 9fadb6011a1ac800a620a83a761bacd468bc2d8a to address it.